packages/sign_in_with_apple/ios/Classes/SwiftSignInWithApplePlugin.swift

```diff
@@ -111,99 +129,161 @@ public class SwiftSignInWithApplePlugin: NSObject, FlutterPlugin {
 
         authorizationController.delegate = self
         authorizationController.performRequests()
     }
+    
+    private func parseData(data: Data?) -> String? {
+        if let data = data {
+            return String(decoding: data, as: UTF8.self)
+        }
+        
+        return nil
+    }
 
-    public func authorizationController(controller _: ASAuthorizationController, didCompleteWithAuthorization authorization: ASAuthorization) {
+    public func authorizationController(
+        controller _: ASAuthorizationController,
+        didCompleteWithAuthorization authorization: ASAuthorization
+    ) {
         switch authorization.credential {
         case let appleIDCredential as ASAuthorizationAppleIDCredential:
-            let userIdentifier = appleIDCredential.user
-            let fullName = appleIDCredential.fullName
-            let email = appleIDCredential.email
-
             let result: [String: String?] = [
                 "type": "appleid",
-                "userIdentifier": userIdentifier,
-                "givenName": fullName?.givenName,
-                "familyName": fullName?.familyName,
-                "email": email,
-                "identityToken": appleIDCredential.identityToken != nil ? String(decoding: appleIDCredential.identityToken!, as: UTF8.self) : nil,
-                "authorizationCode": appleIDCredential.authorizationCode != nil ? String(decoding: appleIDCredential.authorizationCode!, as: UTF8.self) : nil,
+                "userIdentifier": appleIDCredential.user,
+                "givenName": appleIDCredential.fullName?.givenName,
+                "familyName": appleIDCredential.fullName?.familyName,
+                "email": appleIDCredential.email,
+                "identityToken": parseData(data: appleIDCredential.identityToken),
+                "authorizationCode": parseData(data: appleIDCredential.authorizationCode),
             ]
-
-            resultCallback(result)
+            callback(result)
 
         case let passwordCredential as ASPasswordCredential:
             let result: [String: String] = [
                 "type": "password",
                 "username": passwordCredential.user,
                 "password": passwordCredential.password,
             ]
-            resultCallback(result)
+            callback(result)
 
         default:
             // Not getting any credentials would result in an error (didCompleteWithError)
-            resultCallback(
+            callback(
                 FlutterError(
-                    code: "ERR",
+                    code: "unknown-credentials",
                     message: "Unexpected credentials: \(authorization.credential)",
                     details: nil
                 )
             )
         }
     }
 
-    public func authorizationController(controller _: ASAuthorizationController, didCompleteWithError error: Error) {
-        if currentAction == .attemptWithPasswords {
```

Comment:
So is the Flutter side now expected to make 2 request to first query the keychain, and if that fails try the Apple ID? I think we might just keep trying one provider after another, else everyone would be writing the same code in their app (first try keychain, then Apple ID).

But giving the user control over the methods and omit some method entirely is great 👍

Reply:
Do you know if the success callback can be called additionally to the error callback? Is that a valid flow?

I find the multiple requests a bit weird, but I do want to do some testing later today here anyways.

Reply:
> Do you know if the success callback can be called additionally to the error callback? Is that a valid flow?

I don't know that's is prohibited, but it would for sure be unexpected to me.

Reply:
While I was looking into this, I didn't find any good reason for actually providing two auth requests.
Once I provided a password request, I ended up always having an unknown error. Only if I removed the password request, I got the apple sign in again.
So I guess this API change should be fine and the end-users could decide which request they want to make when.